### PR TITLE
HPCC-23347 ESP won't generate sample XML for all methods in a service

### PR DESCRIPTION
**Pull request by lucasmauro**
## Type of change:
- [X] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [X] My code follows the code style of this project.
  - [X] My code does not create any new warnings from compiler, build system, or lint.
- [X] The commit message is properly formatted and free of typos.
  - [X] The commit message title makes sense in a changelog, by itself.
  - [X] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [X] I have read the CONTRIBUTORS document.
- [X] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [X] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [X] This change fixes the problem, not just the symptom
  - [X] The target branch of this pull request is appropriate for such a change.
- [X] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [X] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@asselitx Please review

**Comment by lucasmauro** [comment]
@afishbeck if you could also review, please.

File: esp/bindings/http/platform/httpbinding.cpp [hunk 1]
```diff
@@ -1935,30 +1935,41 @@ void EspHttpBinding::generateSampleXmlFromSchema(bool isRequest, IEspContext &co
...
-    if (schema.get())
+    StringBuffer content("<?xml version=\"1.0\" encoding=\"UTF-8\"?>");
+    if (context.queryRequestParameters()->hasProp("display"))
+        content.append("<?xml-stylesheet type=\"text/xsl\" href=\"/esp/xslt/xmlformatter.xsl\"?>");
+
+    if (!methodQName || !*methodQName)
     {
-        IXmlType* type = schema->queryElementType(element);
-        if (type)
+        StringBuffer label(isRequest ? "Request" : "Response");
```
Comment (afishbeck) [inline esp/bindings/http/platform/httpbinding.cpp:1944]: In the native services these tags are plural "Requests", "Responses".

File: esp/bindings/http/platform/httpbinding.cpp [hunk 2]
```diff
@@ -1935,30 +1935,41 @@ void EspHttpBinding::generateSampleXmlFromSchema(bool isRequest, IEspContext &co
...
+        content.appendf("<Examples><%s>", isRequest ? "Request" : "Response");
```
Comment (afishbeck) [inline esp/bindings/http/platform/httpbinding.cpp:1945]: Might as well use label.str() here.

**Review by afishbeck — CHANGES_REQUESTED** [review]
@lucasmauro  minor changes, otherwise looks great.

File: esp/bindings/http/platform/httpbinding.cpp (hunk 1 above)
Comment (lucasmauro) [reply esp/bindings/http/platform/httpbinding.cpp:1944]: The JSON generation returns "Request" and "Response" (line 1912). Should we change them as well?

File: esp/bindings/http/platform/httpbinding.cpp (hunk 2 above)
Comment (lucasmauro) [reply esp/bindings/http/platform/httpbinding.cpp:1945]: Thanks!

File: esp/bindings/http/platform/httpbinding.cpp (hunk 1 above)
Comment (afishbeck) [reply esp/bindings/http/platform/httpbinding.cpp:1944]: Ah, I see that.  In some ways it's inconsistent, but yes, in native services it is plural in xml, and singular in JSON.  So I would make it plural here.

Btw, the reason is that in xml it is a root element that contains muliple children of different names.  So <requests><a/><b/><c/>  vs. In JSON it is an array. request[1], request[2].

Comment (lucasmauro) [reply esp/bindings/http/platform/httpbinding.cpp]: Understood, @afishbeck. Thanks for the explanation. I have just pushed your suggested changes.

**Review by afishbeck — APPROVED** [review]
@lucasmauro looks good.

**Pull request merged**